config/spotbugs-exclude.xml

```diff
@@ -280,4 +280,10 @@
     name="com.puppycrawl.tools.checkstyle.site.XdocsTemplateParser"/>
     <Bug pattern="STT_TOSTRING_STORED_IN_FIELD"/>
   </Match>
+  <Match>
+    <!-- We cast to XdocSink to avoid unwanted newlines.
+         Until https://github.com/checkstyle/checkstyle/issues/13426 -->
+    <Class name="com.puppycrawl.tools.checkstyle.site.ViolationMessagesMacro"/>
+    <Bug pattern="PDP_POORLY_DEFINED_PARAMETER"/>
```

Comment:
Please share error message. To let me see on what code id complaining on.
Can we use more fine grained suppression?

Reply:
> [ERROR] Medium: Method com.puppycrawl.tools.checkstyle.site.ViolationMessagesMacro.createListItem(Sink, Class, String) defines parameters more abstractly than needed to function properly [com.puppycrawl.tools.checkstyle.site.ViolationMessagesMacro] At ViolationMessagesMacro.java:[line 83] PDP_POORLY_DEFINED_PARAMETER

```java
        final XdocSink xdocSink = (XdocSink) sink;
```

Reply:
> PDP_POORLY_DEFINED_PARAMETER This method defines parameters at a more abstract level than is actually needed to function correctly, as the code casts these parameters to more concrete types. Since this method is not derivable, you should just define the parameters with the type that is needed.

https://fb-contrib.sourceforge.net/bugdescriptions.html

Reply:
`createListItem((XdocSink)sink,....)`